lib/s3routes/routes/routeDELETE.ts

```diff
@@ -54,11 +54,11 @@ export default function routeDELETE(
         api.callApiMethod('objectDelete', request, response, log,
             (err, corsHeaders) => {
                 /*
-              * Since AWS expects a 204 regardless of the existence of
-              the object, the errors NoSuchKey and NoSuchVersion should not
-              * be sent back as a response.
-              */
-                if (err && !err.is.NoSuchKey && !err.is.NoSuchVersion) {
+                 * Since AWS expects a 204 regardless of the existence of
+                 * the object, the errors NoSuchKey and NoSuchVersion should not
+                 * be sent back as a response.
+                 */
+                if (err instanceof ArsenalError && !err.is?.NoSuchKey && !err.is?.NoSuchVersion) {
                     return routesUtils.responseNoBody(err, corsHeaders,
```

Comment:
Should the non arsenal errors also be sent as a response ?

Reply:
No, because they would likely translate code-related error, leaking information about the internals of the application. Better returning standard errors in this case. But the routes should still allow these non-arsenal errors to be received, before sending the final response